test/apps/iodemo/ucx_wrapper.h

```diff
@@ -184,16 +204,28 @@ class UcxConnection {
     bool recv_data(void *buffer, size_t length, uint32_t sn,
                    UcxCallback* callback = EmptyCallback::get());
 
+    bool send_am(const void *meta, size_t meta_length,
+                 const void *buffer, size_t length,
+                 UcxCallback* callback = EmptyCallback::get());
+
+    bool recv_am_data(void *buffer, size_t length,
+                      UcxCallback* callback = EmptyCallback::get());
+
     void cancel_all();
 
-    uint32_t id() const {
+    uint64_t id() const {
         return _conn_id;
     }
 
     ucs_status_t ucx_status() const {
         return _ucx_status;
     }
 
+    void set_am_desc(void *data, const ucp_am_recv_param_t *param) {
+        _am_desc.data       = data;
+        _am_desc.recv_param = param;
```

Comment:
maybe set corresponding handler depends on param from CB?

Reply:
can you please clarify?

Reply:
handle erger/rndv brach from callback and store pointer to eager/rndv handler instead of param.

Reply:
why is it better?

Reply:
we can control `data` pointer by return code in callback, `param` - can't at all, so better to avoid passing it to everywhere else